src/Couchbase.Lite.Shared/Sync/ReplicatorOptionsDictionary.cs

```diff
@@ -49,12 +49,20 @@ internal sealed class ReplicatorOptionsDictionary : OptionsDictionary
         private const string ProtocolsOptionKey = "WS-Protocols";
         private const string RemoteDBUniqueIDKey = "remoteDBUniqueID";
         private const string ResetKey = "reset";
+        private const string Level = "progress";
         private const string Tag = nameof(ReplicatorOptionsDictionary);
 
         #endregion
 
         #region Properties
 
+        [CanBeNull]
+        public ReplicatorOptionProgressLevel ProgressLevel
+        {
+            get => this.GetCast<ReplicatorOptionProgressLevel>(Level);
+            set => this[Level] = value;
```

Comment:
Do the int conversion here instead of in FLEncode

Reply:
change to use int conversion.

Reply:
I don't see the actual enum here anymore, but what I was thinking of was something like this

```
internal ReplicatorProgressLevel ProgressLevel
{
    get => (ReplicatorProgressLevel)this.GetCast<int>(LevelKey);
    set => this[LevelKey] = (int)value;
}
```